.github/workflows/docs-deploy.yml

```diff
@@ -0,0 +1,114 @@
+name: Docs deploy
+concurrency: docs-deploy-${{ github.ref }}
+on:
+  workflow_run:
+    workflows: ["Docs Build"]
+    types:
+      - completed
+
+env:
+  html_folder: ./doc/_build/html
+  artifact_name: doc_html
+
+jobs:
+  # build a redirect index.html to the stable version on tag builds
+  # always run this job (even when it is not needed) to be able to
+  # depend on it in the deploy job
+  redirect-stable:
+    if: >
+      github.repository == 'pyxem/pyxem' &&
+      github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build index redirect
+        if: github.ref_type == 'tag' 
+        env:
+          VERSION: ${{ github.ref_name }}
+        run: |
+          mkdir -p redirect
+          cat <<EOF > redirect/index.html
+          <!DOCTYPE html>
+          <html>
+            <head>
+              <meta http-equiv="refresh" content="0; url=${VERSION}/">
+              <script>window.location.href='${VERSION}/';</script>
+            </head>
+            <body>
+              <p>Redirecting to the <a href="${VERSION}/">documentation (stable)</a>...</p>
+            </body>
+          </html>
+          EOF
+
+      - uses: actions/upload-artifact@v4
+        if: github.ref_type == 'tag' 
+        with:
+          path: redirect
+          name: redirect
+
+  deploy:
+    if: >
+      github.repository == 'pyxem/pyxem' &&
+      github.event.workflow_run.conclusion == 'success'
+    needs: redirect-stable
+    runs-on: ubuntu-latest
+    permissions:
+      # needs write permission to push the docs
+      contents: write
+    steps:
+      - name: Download built docs
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.artifact_name }}
+          path: ${{ env.html_folder }}
+
+      - name: Print doc files
+        run: ls -l ${{ env.html_folder }}
+
+      - name: Deploy dev
+        # run on main branch only
+        if: github.event.workflow_run.head_branch == 'main' 
+        uses: peaceiris/actions-gh-pages@4f9cc6602d3f66b9c108549d475ec49e8ef4d45e
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_branch: gh-pages
+          publish_dir: ${{ env.html_folder }}
+          destination_dir: dev
+
+      - name: Deploy PR preview
+        # run on pull request only
+        if: github.event_name == 'pull_request'
+        uses: rossjrw/pr-preview-action@9f77b1d057b494e662c50b8ca40ecc63f21e0887
+        with:
+          source-dir: ${{ env.html_folder }}
+          deploy-repository: CSSFrancis/pyxem-docs-staging
+          token: ${{ secrets.PREVIEW_DEPLOY_KEY }}
+          comment: true
+
+      - name: Deploy versioned docs
+        # run on tag only
+        if: github.ref_type == 'tag'
```

Comment:
This should be on a tag or a workflow dispatch. 

Just if we make a new release and the documentation build fails then the idea is that we can just run the workflow dispatch to build a new version of the docs.  That and we can overwrite a version if necessary.

Reply:
I don't think that it is necessary to specify a tag (this was actually a comment that I had when reviewing #1164) because I expect that it will work as follow:
1. select a tag from the "Run workflow" menu - see screenshot below
2. `github.ref_type` should be `tag` and therefore the condition for that step true.

Maybe something to add the documentation?

<img width="294" height="263" alt="image" src="https://github.com/user-attachments/assets/cb67d8ce-80df-463d-87f9-9149807cdc43" />

Of course, this is not to say that they may some teething issue with this refactor, but this is the idea!